File: util/src/entities/Webhook.ts
import { Column, Entity, JoinColumn, ManyToOne, RelationId } from "typeorm";
import { Application } from "./Application";
import { BaseClass } from "./BaseClass";
import { Channel } from "./Channel";
import { Guild } from "./Guild";
import { User } from "./User";

export enum WebhookType {
	Incoming = 1,
	ChannelFollower = 2,
}

@Entity("webhooks")
export class Webhook extends BaseClass {
	@Column({ type: "int" })
	type: WebhookType;

	@Column({ nullable: true })
	name?: string;

	@Column({ nullable: true })
	avatar?: string;

	@Column({ nullable: true })
	token?: string;

	@Column({ nullable: true })
	@RelationId((webhook: Webhook) => webhook.guild)
	guild_id: string;

	@JoinColumn({ name: "guild_id" })
	@ManyToOne(() => Guild, {
		onDelete: "CASCADE",
	})
	guild: Guild;

	@Column({ nullable: true })
	@RelationId((webhook: Webhook) => webhook.channel)
	channel_id: string;

	@JoinColumn({ name: "channel_id" })
	@ManyToOne(() => Channel, {
		onDelete: "CASCADE",
	})
	channel: Channel;

	@Column({ nullable: true })
	@RelationId((webhook: Webhook) => webhook.application)
	application_id: string;

	@JoinColumn({ name: "application_id" })
	@ManyToOne(() => Application, {
		onDelete: "CASCADE",
	})
	application: Application;

	@Column({ nullable: true })
	@RelationId((webhook: Webhook) => webhook.user)
	user_id: string;

	@JoinColumn({ name: "user_id" })
	@ManyToOne(() => User, {
		onDelete: "CASCADE",
	})
	user: User;

	@Column({ nullable: true })
	@RelationId((webhook: Webhook) => webhook.guild)
	source_guild_id: string;

	@JoinColumn({ name: "source_guild_id" })
	@ManyToOne(() => Guild, {
		onDelete: "CASCADE",
	})
	source_guild: Guild;
}

File: extra/admin-api/Models/Spacebar.Models.Db/Models/Webhook.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Models.Db.Models;

[Table("webhooks")]
public partial class Webhook
{
    [Key]
    [Column("id")]
    public long Id { get; set; }

    [Column("type")]
    public int Type { get; set; }

    [Column("name", TypeName = "character varying")]
    public string? Name { get; set; }

    [Column("avatar", TypeName = "character varying")]
    public string? Avatar { get; set; }

    [Column("token", TypeName = "character varying")]
    public string? Token { get; set; }

    [Column("guild_id")]
    public long? GuildId { get; set; }

    [Column("channel_id")]
    public long? ChannelId { get; set; }

    [Column("application_id")]
    public long? ApplicationId { get; set; }

    [Column("user_id")]
    public long? UserId { get; set; }

    [Column("source_guild_id")]
    public long? SourceGuildId { get; set; }

    [Column("source_channel_id")]
    public long? SourceChannelId { get; set; }

    [ForeignKey("ApplicationId")]
    [InverseProperty("Webhooks")]
    public virtual Application? Application { get; set; }

    [ForeignKey("ChannelId")]
    [InverseProperty("WebhookChannels")]
    public virtual Channel? Channel { get; set; }

    [ForeignKey("GuildId")]
    [InverseProperty("WebhookGuilds")]
    public virtual Guild? Guild { get; set; }

    [InverseProperty("Webhook")]
    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();

    [ForeignKey("SourceChannelId")]
    [InverseProperty("WebhookSourceChannels")]
    public virtual Channel? SourceChannel { get; set; }

    [ForeignKey("SourceGuildId")]
    [InverseProperty("WebhookSourceGuilds")]
    public virtual Guild? SourceGuild { get; set; }

    [ForeignKey("UserId")]
    [InverseProperty("Webhooks")]
    public virtual User? User { get; set; }
}
